pkg/scheduler/framework/v1alpha1/registry_test.go

```diff
@@ -104,3 +104,185 @@ func loadConfig(data []byte) (*config.KubeSchedulerConfiguration, error) {
+				return false
+			}
+		} else {
+			return false
+		}
+	}
+
+	return true
+}
+
+type mockNoopPlugin struct{}
+
+func (p *mockNoopPlugin) Name() string {
+	return "MockNoop"
+}
+
+func NewMockNoopPluginFactory() PluginFactory {
+	return func(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+		return &mockNoopPlugin{}, nil
+	}
+}
+
+func TestMerge(t *testing.T) {
+	tests := []struct {
+		name            string
+		primaryRegistry Registry
+		registryToMerge Registry
+		expected        Registry
+		shouldError     bool
+	}{
+		{
+			name: "goodMerge",
+			primaryRegistry: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			registryToMerge: Registry{
+				"pluginFactory2": NewMockNoopPluginFactory(),
+			},
+			expected: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+				"pluginFactory2": NewMockNoopPluginFactory(),
+			},
+			shouldError: false,
+		},
+		{
+			name: "badMergeDuplicateFactory",
+			primaryRegistry: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			registryToMerge: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			expected: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			shouldError: true,
+		},
+	}
+
+	for _, scenario := range tests {
+		err := scenario.primaryRegistry.Merge(scenario.registryToMerge)
+
+		if err == nil && scenario.shouldError == true {
+			t.Errorf("Test %s: Merge() Failed to return Error.", scenario.name)
+			continue
+		}
+
+		if err != nil && scenario.shouldError == false {
+			t.Errorf("Test %s: Merge() Unexpectedly returned an error. Err: %v", scenario.name, err)
+		}
+
+		if !isRegistryEqual(scenario.expected, scenario.primaryRegistry) {
+			t.Errorf("Test %s: Failed to Merge(). Expected %v. Got %v instead.", scenario.name, scenario.expected, scenario.primaryRegistry)
+		}
+	}
+}
+
+func TestRegister(t *testing.T) {
+	tests := []struct {
+		name              string
+		registry          Registry
+		nameToRegister    string
+		factoryToRegister PluginFactory
+		expected          Registry
+		shouldError       bool
+	}{
+		{
+			name:              "goodRegister",
+			registry:          Registry{},
+			nameToRegister:    "pluginFactory1",
+			factoryToRegister: NewMockNoopPluginFactory(),
+			expected: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			shouldError: false,
+		},
+		{
+			name: "badRegisterDuplicatePluginFactory",
+			registry: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			nameToRegister:    "pluginFactory1",
+			factoryToRegister: NewMockNoopPluginFactory(),
+			expected: Registry{
+				"pluginFactory1": NewMockNoopPluginFactory(),
+			},
+			shouldError: true,
+		},
+	}
+
+	for _, scenario := range tests {
+		err := scenario.registry.Register(scenario.nameToRegister, scenario.factoryToRegister)
+
+		if err == nil && scenario.shouldError == true {
+			t.Errorf("Test %s: Register() Failed to return Error.", scenario.name)
```

Comment:
If we follows the Golang subtest convention, it should only output:

```go
t.Errorf("Register() failed blabla...")
```

Could you update the tests in this file? (Including the old tests)